### DISCUSSION TITLE
Viewing/troubleshooting the gRPC Internal Server Logs

### DISCUSSION THREAD

**Discussion by pacog1994** [opentdf/platform#938] [General]
Hi Community,

I just wanted to gather information on how to approach troubleshooting the OpenTDF gRPC internal server logs. For context, I'm trying to debug the contents of entityresolution.go > keycloak_entity_resolution.go:EntityResolution. I've had luck debugging the platform and otdfctl using VSC Go debugger by attaching the debugger to one of the running processes but I don't know if I can do the same with a server that isn't exposed.

I've attempted to use grpcurl but so far, no luck.

When I look at the entityresolution.go file, I see there is an `slog` lib:
![image](https://github.com/opentdf/platform/assets/15972323/4d7998f8-c620-40b7-bab4-3ad37be65390)
Can I utilize this somehow? Maybe viewing it from my command line like `cat server.log <--- where this is slog file`

Any help would be appreciated thanks.

**Comment by jrschumacher** [comment]
@pacog1994 do you have this property in your yaml? The `level` can be `debug`, `info`, `warn`, and `error`.

```yaml
logger:
  level: debug
  type: text
  output: stdout
```

**Comment by pacog1994** [comment]
Closing this old discussion post, I eventually solved the issue. My VSC debugger had a 2nd debugger session that was causing problems. I killed the process and the issue was fixed.
